pipeline/backend/kubernetes/pod.go

```diff
@@ -378,12 +374,10 @@ func startPod(ctx context.Context, engine *kube, step *types.Step) (*v1.Pod, err
 	if err != nil {
 		return nil, err
 	}
-
-	pod, err := mkPod(engine.config.Namespace, podName, step.Image, step.WorkingDir, engine.goos, step.BackendOptions.Kubernetes.ServiceAccountName,
-		step.Pull, step.Privileged,
-		step.Commands, step.Volumes, engine.config.ImagePullSecretNames,
-		engine.config.PodLabels, engine.config.PodAnnotations, step.Environment, step.BackendOptions.Kubernetes.NodeSelector,
-		step.ExtraHosts, step.BackendOptions.Kubernetes.Tolerations, step.BackendOptions.Kubernetes.Resources, step.BackendOptions.Kubernetes.SecurityContext, engine.config.SecurityContext)
+	pod, err := mkPod(step, &step.BackendOptions.Kubernetes, engine.config.Namespace, podName, engine.goos,
+		engine.config.ImagePullSecretNames,
+		engine.config.PodLabels, engine.config.PodAnnotations,
```

Comment:
This comment, probably, is not relevant to this PR, but we should make this config immutable. Seems, we should make and pass a copy.
I've got an issue because of that. In #3123 I have the code below:
```go
	if securityContext != nil {
		key, value := apparmorAnnotation(name, securityContext.ApparmorProfile)
		if key != nil && value != nil {
			meta.Annotations[*key] = *value
		}
	}
```
This adds annotation in Agent's configuration. When I run pipelines without security context, then this annotation still exist and pod fails.
I can make a copy of map, but think, that making whole config copy is better.

Reply:
golang has no concept of native immutable maps :/

Reply:
well there are third-partys ...

https://github.com/benbjohnson/immutable
https://github.com/fishy/go-immutable
...
but if they work for us, i dont know. as we have to de-/serialize from/to json inbetween

Reply:
moved discussion into https://github.com/woodpecker-ci/woodpecker/discussions/3165